articles/ros2_dds_security.md

```diff
@@ -0,0 +1,220 @@
+
+The type of mode desired can be specified by setting the `$ROS_SECURITY_STRATEGY` environment variable to "Enforce" (case-sensitive) for strict mode, and anything else for permissive mode.
+
+
+### Support for a master "on/off" switch for all SROS 2 features
+
+In addition to the supported features just discussed, RCL also supports a master shutoff for security features for easy experimentation.
+If it's turned off (the default), none of the security features will be enabled.
+
+In order to enable SROS 2, set the `$ROS_SECURITY_ENABLE` environment variable to "true" (case-sensitive).
+To disable, set to any other value.
+
+
+## Features in the SROS 2 CLI
+
+Configuring a ROS 2 system to be secure in RCL involves a lot of new technology (PKI, DDS governance and permissions files and their syntax, etc.).
+If a user is comfortable with these technologies, the above information should be all that's necessary to properly lock things down.
+However, the [SROS 2 CLI](https://github.com/ros2/sros2) includes a number of commands to help those who don't want to set it all up themselves.
+This section is not meant to be a tutorial, but a reference of capabilities that are available.
+
+
+    $ ros2 security create_keystore <keystore path>
+
+This creates the initial directory structure that will be the `$ROS_SECURITY_ROOT_DIRECTORY`, and also creates a self-signed CA and uses it to sign a governance file that will encrypt all DDS traffic.
+
+
+    $ ros2 security create_key <keystore path> <fully-qualified node instance name>
+
+This creates a new identity for the node, generating a keypair and signing its x.509 certificate using the CA created in `create_keystore`.
+It also creates an initial permissions document (that allows everything by default) and signs it with the same CA.
+It then copies in the governance file created in `create_keystore` as well as the x.509 certificate for the CA to act as both the identity CA as well as the permissions CA.
+
+
+    $ ros2 security generate_policy <policy file>
+
+This takes the current state of the ROS graph (i.e. all the nodes currently running, along with their topic publications/subscriptions, services, etc.) and turns it into an [SROS 2 policy file](/articles/sros2_policy_files.html).
+
+
+    $ ros2 security create_permission \
+        <keystore path> \
+        <fully-qualified node instance name> \
+        <policy file>
+
+This takes an SROS 2 policy file and uses it to generate (and sign) the DDS permissions document for the given node instance.
+
+
+    $ ros2 security generate_artifacts \
+        -k <keystore path> \
+        -n <space-separated list of fully-qualified node instance names> \
+        -p <space-separated list of policy files>
+
+This is essentially a combination of `create_key` and `create_permission`, taking a set of fully-qualified node instance names as well as a set of policy files and generating the entire set of keys and permissions necessary for them to run securely.
```

Comment:
This is also a combination of `create_keystore` if the one provided in cmd line or via environment variable doesn't exist or is not valid

Reply:
Ah! I didn't actually know that, thank you.

Reply:
Updated.